apps/hash-graph/bins/cli/src/main.rs

```diff
@@ -26,31 +24,11 @@ fn main() -> Result<(), GraphError> {
     validation::error::install_error_stack_hooks();
 
     let Args {
-        sentry_dsn,
-        sentry_environment,
         subcommand,
+        tracing_config,
     } = Args::parse_args();
 
-    // Initialize Sentry
-    // When initializing Sentry, a `Drop` guard is returned, once dropped any remaining events are
-    // flushed. This means we need to keep the guard around for the entire lifetime of the program.
-    let _sentry = sentry::init(sentry::ClientOptions {
-        dsn: sentry_dsn,
-        release: sentry::release_name!(),
-        session_mode: sentry::SessionMode::Request,
-        traces_sampler: Some(Arc::new(|ctx| {
-            if Some(true) == ctx.sampled() {
-                1.0
-            } else if ctx.operation() == "http.server" {
-                0.1
-            } else {
-                1.0
-            }
-        })),
-        environment: Some(Cow::Owned(sentry_environment.to_string())),
-
-        ..Default::default()
-    });
-
-    subcommand.execute()
+    let _sentry_guard = init_sentry(&tracing_config.sentry, release_name!());
+
+    subcommand.execute(tracing_config)
```

Comment:
Why are you initializing the logger not here, but instead in `block_on`? I guess because of completions?

Reply:
The initialization of OTLP requires a Tokio runtime. I made the function `async` to enforce this: https://github.com/hashintel/hash/pull/3893/commits/705f399075a9e5cafa111cd44de5e91f48289259